src/controller/tests/TestWriteChunking.cpp

```diff
@@ -358,8 +515,96 @@ TEST_F(TestWriteChunking, TestBadChunking)
 }
 
 /*
- * When chunked write is enabled, it is dangerious to handle multiple write requests at the same time. In this case, we will reject
- * the latter write requests to the same attribute.
+ * A Variant of TestBadChunking above, that tests the Code Path where we encode a Non-Replace All List in WriteRequests, this
+ * happens with the ACL Cluster.
+ */
+TEST_F(TestWriteChunking, TestBadChunking_NonEmptyReplaceAllList)
+{
+    auto sessionHandle = GetSessionBobToAlice();
+
+    bool atLeastOneRequestSent   = false;
+    bool atLeastOneRequestFailed = false;
+
+    // Initialize the ember side server logic
+    app::InteractionModelEngine::GetInstance()->SetDataModelProvider(CodegenDataModelProviderInstance(nullptr /* delegate */));
+    InitDataModelHandler();
+
+    // Register our fake dynamic endpoint.
+    emberAfSetDynamicEndpoint(0, kTestEndpointId, &testEndpointAcl, Span<DataVersion>(dataVersionStorageAcl));
+
+    // Register our fake attribute access interface.
+    AttributeAccessInterfaceRegistry::Instance().Register(&testServerAcl);
+
+    app::AttributePathParams attributePath(kTestEndpointId, AccessControl::Id, AccessControl::Attributes::Acl::Id);
+
+    constexpr uint8_t kTestListLengthBadChunking = 5;
+
+    for (int bufferSize = 850; bufferSize < static_cast<int>(chip::app::kMaxSecureSduLengthBytes); bufferSize++)
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        TestWriteCallback writeCallback;
+
+        ChipLogDetail(DataManagement, "Running iteration with OCTET_STRING length = %d\n", bufferSize);
+
+        app::WriteClient writeClient(&GetExchangeManager(), &writeCallback, Optional<uint16_t>::Missing());
```

Comment:
nit, would `NullOptional` work?

If so can you please replace all `Optional<uint16_t>::Missing()` that you are adding in this file?

```suggestion
        app::WriteClient writeClient(&GetExchangeManager(), &writeCallback, NullOptional);
```